chapters/inheritance.tex

```diff
@@ -983,60 +983,107 @@ \subsection{Restrictions on Redeclarations}\label{restrictions-on-redeclarations
+\end{semantics}
+\end{annotationdefinition}
 
 \begin{example}
+Using \lstinline!choices! to suggest suitable redeclarations:
 \begin{lstlisting}[language=modelica]
 replaceable model MyResistor = Resistor
   annotation(choices(
-               choice(redeclare model MyResistor=lib2.Resistor(a={2}) "$\ldots$"),
-               choice(redeclare model MyResistor=lib2.Resistor2 "$\ldots$")));
+    choice(redeclare model MyResistor = lib2.Resistor(a = {2}) "$\ldots$"),
+    choice(redeclare model MyResistor = lib2.Resistor2 "$\ldots$")
+  ));
 
 replaceable Resistor Load(R = 2) constrainedby TwoPin
   annotation(choices(
-               choice(redeclare lib2.Resistor Load(a={2}) "$\ldots$"),
-               choice(redeclare Capacitor Load(L=3) "$\ldots$")));
+    choice(redeclare lib2.Resistor Load(a = {2}) "$\ldots$"),
+    choice(redeclare Capacitor Load(L = 3) "$\ldots$")
+  ));
 
 replaceable FrictionFunction a(func = exp) constrainedby Friction
   annotation(choices(
-               choice(redeclare ConstantFriction a(c=1) "$\ldots$"),
-               choice(redeclare TableFriction a(table="$\ldots$") "$\ldots$"),
-               choice(redeclare FunctionFriction a(func=exp) "$\ldots$")));
-
-replaceable package Medium = Modelica.Media.Water.ConstantPropertyLiquidWater
-  constrainedby Modelica.Media.Interfaces.PartialMedium
-  annotation(choicesAllMatching = true);
+    choice(redeclare ConstantFriction a(c = 1) "$\ldots$"),
+    choice(redeclare TableFriction a(table = "$\ldots$") "$\ldots$"),
+    choice(redeclare FunctionFriction a(func = exp) "$\ldots$")
+  ));
 \end{lstlisting}
+\end{example}
 
-It can also be applied to nonreplaceable declarations, e.g.\ to describe enumerations.
+\begin{example}
+Using \lstinline!choices! to suggest suitable value modifications for a non-replaceable declaration:
 \begin{lstlisting}[language=modelica]
 type KindOfController = Integer(min = 1, max = 3)
   annotation(choices(
-              choice = 1 "P",
-              choice = 2 "PI",
-              choice = 3 "PID"));
+    choice = 1 "P",
+    choice = 2 "PI",
+    choice = 3 "PID"
+  ));
 
 model A
   parameter KindOfController x;
 end A;
+
 A a(x = 3 "PID");
 \end{lstlisting}
+\end{example}
 
-It can also be applied to \lstinline!Boolean! variables to define a check box.
+\begin{example}
+Using \lstinline!choices! to request that a checkbox is used to select modification for a \lstinline!Boolean! variable:
 \begin{lstlisting}[language=modelica]
 parameter Boolean useHeatPort = false annotation(choices(checkBox = true));
 \end{lstlisting}
 \end{example}
+
+\begin{annotationdefinition}[choicesAllMatching]
+\begin{synopsis}[grammar]\begin{lstlisting}
+"choicesAllMatching" "=" ( true | false )
+\end{lstlisting}\end{synopsis}
```

Comment:
OK, let's start with something relatively simple, like this one.

I suppose an alternative starting point would be this:
```
Boolean choicesAllMatching
```

To express that we just want constant values, should we use `constant`?
```
constant Boolean choicesAllMatching
```

Should we on top of this write in the text that the value of the annotation should be a literal, or should we introduce some new pseudo-code way of saying this systematically?
```
literal Boolean choicesAllMatching
```

Reply:
(Still, I'd much rather have separate discussions about how to deal with various annotations that we cannot express precisely using today's pseudo-code style.  In other words, go with grammar for now, only to keep it until we have invented something else allowing us to present annotations systematically and without loss of precision.)

Reply:
I agree that it's worth opening a separate issue to bring up for discussion. Sorry, I totally did not catch that the idea was to update the presentation of tables for annotations. I will have to review it all again under that assumption and ignore the way annotations are written.

Reply:
Resolving, since it was decided to move it into a separate discussion.